app/src/test/java/com/example/parkingproject/mvp/presenter/ReleasePresenterTest.java

```diff
@@ -0,0 +1,185 @@
+import com.example.parkingproject.mvp.model.ReleaseModel;
+import com.example.parkingproject.mvp.view.ReleaseView;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Date;
+
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class ReleasePresenterTest {
+    private ReleasePresenter presenter;
+    private ReleaseModel model;
+    private ReleaseView view;
+
+    @Before
+    public void setup() {
+        model = mock(ReleaseModel.class);
+        view = mock(ReleaseView.class);
+        presenter = new ReleasePresenter(model, view);
+    }
+
+    //region onReleaseParkingPressed tests
+    @Test
+    public void onReleaseParkingPressed_success() throws ReleaseMoreThanOneMatchException, ReleaseNoMatchException, ReservationListEmptyException {
+        ReleasePresenter spyPresenter = spy(presenter);
+
+        when(view.getParkingLotNumber()).thenReturn("15");
+        when(view.getSecurityCode()).thenReturn("A123");
+        when(model.getParkingNumber("15")).thenReturn(15);
+
+        Parking parking = new Parking(16);
+        Reservation reservation = new Reservation("A123", new Date().getTime(), new Date().getTime(), 15);
+        parking.getReservationsList().add(reservation);
+
+        when(model.getParking()).thenReturn(parking);
+
+        doReturn(true).when(spyPresenter).validateSecurityCode("A123");
+        doReturn(true).when(spyPresenter).validateParkingLot(15);
+
+        spyPresenter.onReleaseParkingPressed();
+
+        verify(model).parkingRelease(reservation);
+        verify(view).showSuccessMessageReleaseReservation();
+    }
+
+    @Test
+    public void onReleaseParkingPressed_reservationListEmptyException_showError() throws ReleaseMoreThanOneMatchException, ReleaseNoMatchException, ReservationListEmptyException {
+        ReleasePresenter spyPresenter = spy(presenter);
+
+        when(view.getParkingLotNumber()).thenReturn("15");
+        when(view.getSecurityCode()).thenReturn("A123");
+        when(model.getParkingNumber("15")).thenReturn(15);
+
+        Parking parking = new Parking(16);
+        Reservation reservation = new Reservation("A123", new Date().getTime(), new Date().getTime(), 15);
+
+        when(model.getParking()).thenReturn(parking);
+
+        doThrow(new ReservationListEmptyException()).when(model).parkingRelease(reservation);
+
+        spyPresenter.onReleaseParkingPressed();
+
+        verify(model).parkingRelease(reservation);
+        verify(view).showErrorReservationListEmptyException();
+    }
+
+    @Test
+    public void onReleaseParkingPressed_releaseMoreThanOneMatchException_showError() throws ReleaseMoreThanOneMatchException, ReleaseNoMatchException, ReservationListEmptyException {
```

Comment:
I think you can remove the throws part.

throws ReleaseMoreThanOneMatchException, ReleaseNoMatchException, ReservationListEmptyException